### PR TITLE
Fixing doc build workflow

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #314

Fixing the issue [mentioned here](https://github.com/pytorch/data/pull/290#issuecomment-1075241231).

I have ran it on my fork and the build worked:
https://github.com/NivekT/data/actions/runs/2023027716

Differential Revision: [D35048050](https://our.internmc.facebook.com/intern/diff/D35048050)

### CODE REVIEW

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35048050).

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -20,7 +20,8 @@ jobs:
           python3 -m pip install setuptools
           python3 -m pip install --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
           python3 -m pip install matplotlib
-          python3 -m pip install --user "git+https://github.com/pytorch/data.git"
+          python3 -m pip install -r https://raw.githubusercontent.com/pytorch/data/main/requirements.txt
+          python3 -m pip install --no-build-isolation git+https://github.com/pytorch/data
```
Comment (ejguan) [inline .github/workflows/docs.yml:24]: You can add a separate step to compile torchdata from local source as we have a prior step which has already checked out torchdata repo.
```
python3 -m pip install -r requirements.txt
python3 setup.py install
```

**Review by ejguan — APPROVED** [review]
Thanks for fixing it. A nit comment to make this workflow better.

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35048050).

**Pull request closed without merging**